### PR TITLE
PLONK benchmarks

### PR DESCRIPTION
**Pull request by ebfull**
Also adds metrics collection for performance modelling.

Part of #47.

### CODE REVIEW

**Review by therealyingtong — APPROVED** [review]
ACK

**Review by daira — APPROVED** [review]
utACK, but this will need to be rebased on top of #54.

File: examples/performance_model.rs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    // TODO: Make dynamic.
+    let k = 11;
+
+    // Initialize the polynomial commitment parameters
+    let params: Params<EqAffine> = Params::new::<DummyHash<Fq>>(k);
+
+    let empty_circuit: MyCircuit<Fp> = MyCircuit { a: None, k };
+
+    // Initialize the proving key
+    let pk = keygen(&params, &empty_circuit).expect("keygen should not fail");
```
Comment (daira) [inline examples/performance_model.rs:227]: Split this out and collect metrics for it separately.

**Comment by str4d** [comment]
Rebased to fix merge conflicts with #60.

File: examples/performance_model.rs [hunk 2]
```diff
@@ -230,21 +259,35 @@ fn main() {
...
     };
 
     // Create a proof
-    let proof = Proof::create::<DummyHash<Fq>, DummyHash<Fp>, _>(&params, &pk, &circuit, &[])
-        .expect("proof generation should not fail");
+    let proof = Proof::create::<DummyHash<Fq>, DummyHash<Fp>, _>(
+        &params,
+        &pk,
+        &circuit,
+        &[pubinputs.clone()],
```
Comment (daira) [inline examples/performance_model.rs:280]: Remove the redundant clone pointed out by clippy.

**Review by daira — APPROVED** [review]
ACK a65c818fa1546d1bb0dceffc6e9018cb9cf5cf7f modulo a minor comment.

**Review by str4d — APPROVED** [review]

**Pull request merged**